ui/src/app/subnets/views/SubnetDetails/SubnetSummary/SubnetSummary.test.tsx

```diff
@@ -0,0 +1,215 @@
+import { render, screen } from "@testing-library/react";
+import { Provider } from "react-redux";
+import { MemoryRouter, Route } from "react-router-dom";
+import configureStore from "redux-mock-store";
+
+import SubnetSummary from "./SubnetSummary";
+
+import subnetsURLs from "app/subnets/urls";
+import {
+  rootState as rootStateFactory,
+  subnet as subnetFactory,
+  subnetState as subnetStateFactory,
+  space as spaceFactory,
+  spaceState as spaceStateFactory,
+  vlan as vlanFactory,
+  vlanState as vlanStateFactory,
+  fabric as fabricFactory,
+  fabricState as fabricStateFactory,
+} from "testing/factories";
+
+const mockStore = configureStore();
+
+const defaultSubnetData = {
+  id: 1,
+  name: "Test subnet",
+  cidr: "192.168.1.1/32",
+  gateway_ip: "192.168.1.1/32",
+  dns_servers: "Test DNS",
+  description: "Test description",
+  managed: true,
+  active_discovery: true,
+  allow_proxy: true,
+  allow_dns: true,
+  space: 1,
+  vlan: 1,
+};
+
+const renderTestCase = (subnet = subnetFactory(defaultSubnetData)) => {
```

Comment:
We used to set up state using a function like this but it made it a lot harder to change the state for different tests so it'd better if this set up state like our other components e.g. https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/subnets/views/VLANDetails/VLANSummary/VLANSummary.test.tsx#L41